cobbler/remote.py

```diff
@@ -440,37 +378,29 @@ def get_event_log(self, event_id: str) -> str:
         :param event_id: The event-id generated by Cobbler.
         :return: The event log or a ``?``.
         """
-        event_id = str(event_id).replace("..", "").replace("/", "")
-        path = "/var/log/cobbler/tasks/%s.log" % event_id
-        self._log("getting log for %s" % event_id)
+        if not isinstance(event_id, str):
+            raise TypeError('"event_id" must be of type str!')
+        if event_id not in self.events:
+            # This ensures the event_id is valid, and we only read files we want to read.
+            return "?"
+        path = f"/var/log/cobbler/tasks/{event_id}.log"
+        self._log(f"getting log for {event_id}")
         if os.path.exists(path):
-            fh = open(path, "r")
-            data = str(fh.read())
-            fh.close()
+            with open(path, "r", encoding="utf-8") as fh:
+                data = str(fh.read())
             return data
         else:
             return "?"
 
-    def __generate_event_id(self, optype: str) -> str:
-        """
-        Generate an event id based on the current timestamp
-
-        :param optype: Append an additional str to the event-id
-        :return: An id in the format: "<4 digit year>-<2 digit month>-<two digit day>_<2 digit hour><2 digit minute>
-                 <2 digit second>_<optional string>"
-        """
-        (year, month, day, hour, minute, second, weekday, julian, dst) = time.localtime()
-        return "%04d-%02d-%02d_%02d%02d%02d_%s" % (year, month, day, hour, minute, second, optype)
-
-    def _new_event(self, name: str):
+    def _new_event(self, name: str) -> utils.CobblerEvent:
         """
         Generate a new event in the in memory event list.
 
         :param name: The name of the event.
         """
-        event_id = self.__generate_event_id("event")
-        event_id = str(event_id)
-        self.events[event_id] = [float(time.time()), str(name), EVENT_INFO, []]
+        new_event = utils.CobblerEvent(name=name, statetime=time.time())
+        self.events[new_event.event_id] = new_event
+        return new_event
 
     def __start_task(self, thr_obj_fn, token: str, role_name: str, name: str, args: dict, on_done=None):
```

Comment:
Would it make sense to rename `args` to `options` here? Seems like that's the naming used throughout the codebase for this (both by code that call as well as use this method)